### ISSUE TITLE
macOS: MessageBoxes attached to a window don't work with VoiceOver (system text to speech)

### ISSUE THREAD

**Issue by slime73** [libsdl-org/SDL#6948]
Repro steps:
* Enable VoiceOver in System Preferences -> Accessibility
* Run an SDL app which calls `SDL_ShowSimpleMessageBox` with a valid window pointer.
* Observe no VoiceOver dialog when the message box appears.
* Run an SDL app which calls `SDL_ShowSimpleMessageBox` with a null window pointer.
* Observe some VoiceOver dialog. I'm not sure it's recognizing button names properly in this situation but I don't know if that's expected.

I haven't investigated very thoroughly yet but I wonder if the SDL NSWindow is either eating some important events or missing something which is needed for VoiceOver to work?

(Issue originally reported here https://github.com/love2d/love/issues/1869)

**Comment by TheQuinbox** [comment]
Hi, just did some testing with VoiceOver, and the dialog reads properly with a null window pointer (i.e. passing false to love.window.showMessageBox()). Default focus is a little weird, for example, getting put on the OK button, not the text in an "info" dialog, but it at least works.

**Comment by slouken** [comment]
Fix merged, thanks!

**Comment by slime73** [comment]
Thanks! Could this also be backported to SDL2?

**Comment by slime73** [comment]
> Thanks! Could this also be backported to SDL2?

This is done now.
